Extensions/RosettaTraining/Scripts/TensorFlow/data_reader.py

```diff
@@ -0,0 +1,218 @@
+import os
+import json
+
+LABEL_IF_PLAYER1_WIN = 1
+LABEL_IF_PLAYER1_LOSS = -1
+
+kLabelIfFirstPlayerLoss = -1
+kLabelIfFirstPlayerWin = 1
+kLabelFirstPlayerWinIfGreaterThan = 0
+
+kHeroFeatures = 1
+kMinionFeatures = 7  # exclude card id
+kMinions = 7
+
+kCurrentHandCards = 10
+kCurrentHandCardFeatures = 1  # exclude card id
+kCurrentHandFeatures = 1
+
+kMaxCardId = 1860
+
+class DynamicMapper:
+    def __init__(self, start_index = 0):
+        self.dict_v_to_index = {}
+        self.dict_index_to_v = {}
+        self.next_index = start_index
+
+    def add_map(self, v):
+        idx = self.next_index
+        self.next_index = idx + 1
+
+        self.dict_v_to_index[v] = idx
+        self.dict_index_to_v[idx] = v
+
+    def get_index(self, v):
+        if v not in self.dict_v_to_index:
+            self.add_map(v)
+        return self.dict_v_to_index[v]
+
+    def get_index_to_v_dict(self):
+        return self.dict_index_to_v
+
+class DataReader:
+    def __init__(self, dirname):
+        self.dirname = dirname
+        self.data = []
+        self.hand_card_mapper = DynamicMapper(1)
+
+    @classmethod
+    def from_bool(cls, v):
+        if v is True:
+            return 1.0
+        else:
+            return -1.0
+
+    def read_hero_data(self, data):
+        health = data['hero']['health']
+        armor = data['hero']['armor']
+
+        self.data.extend([health + armor])
+
+    def add_minion_data_placeholder(self):
+        self.data.extend([
+            0, 0, 0, 0,
+            -1.0, -1.0, -1.0, -1.0
+        ])
+
+    def add_minion_data(self, minion):
+        attackable = False
+        try:
+            attackable = minion['attackable']
+        except Exception as _:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable '_' is assigned to but never used](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4748929)